source/write/replace.txt

```diff
@@ -0,0 +1,282 @@
+.. _cpp-write-replace:
+
+=================
+Replace Documents
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: modify, change, bulk, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to run a replace operation on a MongoDB collection.
+A replace operation removes all fields except the ``_id`` field in the target document and replaces them with new ones.
+You can call the ``replace_one()`` method to replace a single document.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your C++ application, instantiate a ``mongocxx::client`` that connects to an Atlas cluster
+and assign the following values to your ``db`` and ``collection`` variables:
+
+.. literalinclude:: /includes/write/replace.cpp
+    :start-after: start-db-coll
+    :end-before: end-db-coll
+    :language: cpp
+    :dedent:
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see
+:atlas:`Get Started with Atlas </getting-started>`.
+
+Replace Operation
+-----------------
+
+You can perform a replace operation by calling the ``replace_one()`` method.
+This method removes all fields except the ``_id`` field from the first document that
+matches the search criteria. It then inserts the fields and values you specify into the document. 
+
+Each replace method requires the following parameters:
+
+- **Query filter** document: Specifies which document to replace. For
+  more information about query filters, see 
+  :manual:`Query Filter Documents </core/document/#query-filter-documents>` in
+  the {+mdb-server+} manual.
+
+- **Replace** document: Specifies the fields and values to insert in the new document. 
+
+.. important:: 
+
+   The values of ``_id`` fields are immutable. If your replacement document specifies a value for the ``_id`` field, 
+   it must match the ``_id`` value of the existing document.
+
+Replace One Document Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example uses the ``replace_one()`` method to replace 
+a document that has a ``name`` field value of ``Nobu`` with a new document 
+that has a ``name`` field value of ``La Bernadin``:
+
+.. literalinclude:: /includes/write/replace.cpp
+   :start-after: start-replace-one
+   :end-before: end-replace-one
+   :language: cpp
+   :dedent:
+
+To check if you successfully replaced the document, you can use the :ref:`find_one() <cpp-retrieve-find-one>` method to print out the new document:
```

Comment:
S: we try to avoid adding a link to a term without describing where the link goes ([style guide](https://www.mongodb.com/docs/meta/style-guide/style/links-and-cross-references/#make-link-text-clear)). You could remove this inline link and add a tip admonition after the code example that says something like "To learn more about the ``find_one()`` method, see :ref:`cpp-retrieve-find-one` in the Retrieve Data guide."